src/app/chat/chat.service.ts

```diff
@@ -128,29 +160,39 @@ export class ChatService {
    *  size:20
    * }
    */
-  getMessageList(prams: messageListPrams): Observable<any> {
-    console.log("Service", this.messageList);
-    return of(this.messageList);
+  getMessageList(data: messageListPrams): Observable<any> {
+    return this.request.get(api.getChatMessages, {
+      params: data
+    });
   }
 
   markMessagesAsSeen(prams: markAsSeenPrams): Observable<any> {
-    prams.action = 'mark_seen';
-    return of("maked");
+    let body = {
+      'team_id': prams.team_id,
+      'id': prams.id,
+      'action': 'mark_seen'
+    }
+    return this.request.post(api.markAsSeen, body, {
+      headers: { 'Content-Type': 'application/x-www-form-urlencoded' }
```

Comment:
Can we use the default content-type `application/json` for the header of this API call? If so, don't need to specify it here. Same for other places

Reply:
yes,`application/json` can use as default content-type, but in hear this api calls not working if we didn't pass `application/x-www-form-urlencoded` as the content-type. I have try to remove it. in the way request service if we need and header pass with api we need to pass it to request service if we didn't pass any header default header will set to api request. this content-type is only need in some api request so when we calling those api request from service we need to pass this custom content-type to request service to set request header.